legacy_packages/sdk/src/evm/common/utils.ts

```diff
@@ -30,6 +30,27 @@ export function unique<T>(a: T[], fn: (a: T, b: T) => boolean): T[] {
   return a;
 }
 
+/**
+ * @internal
+ */
+export function hasDuplicates<T>(a: T[], fn: (a: T, b: T) => boolean): boolean {
```

Comment:
could we rename the params so that this is more easy to ready (too many a, a, b)

at the least something like
```suggestion
export function hasDuplicates<T>(array: T[], fn: (a: T, b: T) => boolean): boolean {
```